### PR TITLE
bootstrap.effectivemass: Do not rely on original data

### PR DESCRIPTION
**Pull request by martin-ueding**
The `bootstrap.effectivemass` function works on a bootstrapped correlator. The data member `cf` belongs to the `cf_orig` mixin and is not present in general cases. Therefore this will fail. But we can just use `cf0` which has to be present.

### CODE REVIEW

File: R/effectivemass.R [hunk 1]
```diff
@@ -217,10 +217,7 @@ bootstrap.effectivemass <- function(cf, type="solve") {
   stopifnot(inherits(cf, 'cf_boot'))
 
   ## number of measurements
-  N <- length(cf$cf[,1])
-  if(is.null(cf$cf)) {
-    N <- cf$N
-  }
+  N <- length(cf$cf0)
```
Comment (kostrzewa) [inline R/effectivemass.R:220]: this cannot be correct, the number of measurements is not the length of `cf0`

Comment (kostrzewa) [reply R/effectivemass.R:220]: The order of operations on the LHS is simply wrong and probably the right thing to do would be to inherit `cf$N` to begin with.

Comment (martin-ueding) [reply R/effectivemass.R:220]: Oh, you are right. It is the number of measurements and not the number of time slices.

But why does the effective mass need to know the number of measurements anyway?

Comment (martin-ueding) [reply R/effectivemass.R:220]: Oh, it just preserves it. I think it just does not belong in there, just as the original correlator should not be in it. At least we have “effectivemass *has* cf” here, unlike “matrixfit *is* cf”.

Comment (urbach) [reply R/effectivemass.R:220]: but now `N` is not defined anymore? `N` is still part of the returned `list`, isn't it?

**Comment by martin-ueding** [comment]
I see that we have two options: Remove `N` completely from the effective mass, or make `N` a requirement of the `cf_meta` mixin and make sure that it is present *everywhere*.

**Comment by martin-ueding** [comment]
I have removed this now completely as the `cf_meta` does not have this as a specification.

**Pull request merged**